src/components/settings/Account/Billing/ManageButton.tsx

```diff
@@ -24,59 +24,70 @@ export const ManageBillingButton: React.FC<ManageBillingButtonProps> = ({
   buttonProps = { variant: "outline", color: loading ? "gray" : "blue.500" },
   onClick,
 }) => {
-  const [sessionUrl, setSessionUrl] = useState();
+  const [sessionUrl, setSessionUrl] = useState(account.stripePaymentActionUrl);
   const mutation = useCreateBillingSession();
-  const validPayment = account.status === AccountStatus.ValidPayment;
-  const paymentVerification =
-    account.status === AccountStatus.PaymentVerification;
-  const invalidPayment = [
-    AccountStatus.InvalidPayment,
-    AccountStatus.InvalidPaymentMethod,
-  ].includes(account.status);
+
+  let [buttonLabel, buttonText]: string[] = ["", ""];
+  switch (account.status) {
+    case AccountStatus.InvalidPayment:
+    case AccountStatus.ValidPayment: {
+      buttonLabel = "manage";
+      buttonText = "Manage billing";
+      break;
+    }
+    case AccountStatus.PaymentVerification: {
+      buttonLabel = "verifyPaymentMethod";
+      buttonText = "Verify your payment method →";
+      break;
+    }
+    case AccountStatus.InvalidPaymentMethod: {
+      buttonLabel = "addAnotherPayment";
+      buttonText = "Add another payment method →";
+      break;
+    }
+    default: {
+      buttonLabel = "addPayment";
+      buttonText = "Add payment method";
+      break;
+    }
+  }
 
   const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
-    if (onClick || loading || !sessionUrl) {
+    if (loading) {
+      e.preventDefault();
+      return;
+    }
+
+    if (!["verifyPaymentMethod", "manage"].includes(buttonLabel)) {
       e.preventDefault();
-      if (onClick) {
-        onClick();
-      }
+      onClick?.();
     }
   };
 
   useEffect(() => {
-    if (!onClick) {
+    if (buttonLabel === "manage" && !sessionUrl) {
       mutation.mutate(undefined, {
         onSuccess: (data) => {
           setSessionUrl(data.url);
         },
       });
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [onClick]);
+  }, [buttonLabel]);
```

Comment:
this is an extremely weird pattern

I assume the behavior we want here is:

if `buttonLabel` is `manage` and we do not have a `sessionUrl` -> create a session and return the `sessionUrl`

but we've (for some bizarre reason) split this into a `useState` and `useMutation` when what this *should* be is a single `useQuery` that returns a sessionUrl and is `enabled` when buttonLabel === 'manage'

this will absolutely cause bugs down the line, we have to fix this

Reply:
Good feedback, I updated it to do the pattern you suggested. I didn't know how convenient react-query is and it really cleaned up this logic here

Thanks for the idea @jnsdls!